expr/tagquery/expression.go

```diff
@@ -463,3 +463,90 @@ func (o ExpressionOperator) StringIntoWriter(writer io.Writer) {
 		writer.Write([]byte("!="))
 	}
 }
+
+const SeriesByTagIdent = "seriesByTag("
+
+//IsSeriesByTag returns true if the given query is a `seriesByTag` query
+func IsSeriesByTagExpression(query string) bool {
+	return strings.HasPrefix(query, SeriesByTagIdent)
+}
+
+// ParseSeriesByTagExpression takes a `seriesByTag` query which includes multiple tag query expressions
+// example query: "seriesByTag('a=b', 'c=d', 'e!=~f.*')"
+// it then returns a slice of strings where each string is one of the expressions, and an error
+// which is non-nil if there was an error in the expression parsing and validation
+// all expressions get validated and an error is returned if one or more are invalid
+func ParseSeriesByTagExpression(seriesByTagQuery string) (Expressions, error) {
```

Comment:
i think there is a potential that this doesn't return an error on an invalid query string such as `seriesByTag('a=b')xyz`. I'd check that the last char is a `)`. I suspect that a simple `if seriesByTagQuery[len(seriesByTagQuery)-1] == ')'` is also faster than having to call `strings.LastIndex()`

Reply:
or is there a specific reason why we need to allow that there is more stuff after the last `)`? like white spaces?

Reply:
not sure, this is just copy-pasted from the earlier usage.

I think it might actually blow up on input like [this](https://graphite.readthedocs.io/en/latest/functions.html#graphite.render.functions.aliasByNode):
```
&target=seriesByTag("name=~cpu.load.*", "server=~server[1-9]+", "datacenter=dc1")|aliasByNode("datacenter", "server", 1)
# will produce output series like
# dc1.server1.load5, dc1.server2.load5, dc1.server1.load10, dc1.server2.load10
```

But that apparently hasn't been an issue so far.

Reply:
Yeah, I don't think we support the piped syntax in MT yet, that syntax is quite new

Reply:
Updated to check that it ends with a `)` in 67a3a0a6692600734528a2d5b18a54ff775cd139.

There are also some tests now, so at least the behavior is defined

Reply:
Cool, thx. I'd mention in the comment that `ParseSeriesByTagExpression()` does not validate the string, it relies on `IsSeriesByTagExpression()` to validate. But looks good otherwise

Reply:
good call, added in 86c0de89303dc544b88ae26f8f00bc53e46b0bb6